code/game/objects/effects/roaches.dm

```diff
@@ -67,8 +62,11 @@
 			src.loc = O.owner ? O.owner.loc : O.loc
 
 		var/spawn_type = /mob/living/carbon/superior_animal/roach/roachling
-		new spawn_type(src.loc, src)
-		qdel(src)
+		if (isturf(src.loc) || istype(src.loc, /obj/structure/closet)) // suppresses hatching when not in a suitable loc
```

Comment:
I'd make this also stop the eggs from growing so the eggs wouldn't instantly hatch when being dropped from the inventory,